### DISCUSSION TITLE
OpenVAS 3x medium security issue found

### DISCUSSION THREAD

**Discussion by juched78** [nextcloud/all-in-one#3602] [Questions] [answered]
Running OpenVAS on my network, the NextCloud AIO solution was found to have a medium security issue with a cookie use.

Port 8081 (nextcloud/all-in-one:latest) and 11001 (nextcloud-aio-apache)

<html>
<body>
<div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Summary</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">The remote HTTP web server / application is missing to set the
'Secure' cookie attribute for one or more sent HTTP cookie.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Detection Result</h2><div style="box-sizing: inherit;"><div class="sc-gRnDUn dLEdfc" style="box-sizing: inherit; min-width: 500px; max-width: 1080px;"><pre class="sc-drKuOJ bOoNFk" style="box-sizing: inherit; white-space: pre-wrap; overflow-wrap: break-word;">The cookies:

Set-Cookie: PHPSESSID=***replaced***; path=/

are missing the "Secure" cookie attribute.</pre></div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Insight</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">The flaw exists if a cookie is not using the 'Secure' cookie
attribute and is sent over a SSL/TLS connection.

This allows a cookie to be passed to the server by the client over non-secure channels (HTTP) and
subsequently allows an attacker to e.g. conduct session hijacking attacks.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Detection Method</h2><div style="box-sizing: inherit;"><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">Checks all cookies sent by the remote HTTP web server /
application over a SSL/TLS connection for a missing 'Secure' cookie attribute.</div>

Details: | Missing 'Secure' Cookie Attribute (HTTP) OID: 1.3.6.1.4.1.25623.1.0.902661
-- | --
Version used: | 2023-01-17T10:10:58Z

</div></div>
</body>
</html>

### Expected behavior 

No finding reported by OpenVAS.

### Actual behavior 
Finding reported by OpenVAS.

#### Other valuable info 
Summary
The remote HTTP web server / application is missing to set the
'Secure' cookie attribute for one or more sent HTTP cookie.
Detection Result
The cookies:

Set-Cookie: PHPSESSID=***replaced***; path=/

are missing the "Secure" cookie attribute.
Insight
The flaw exists if a cookie is not using the 'Secure' cookie
attribute and is sent over a SSL/TLS connection.

This allows a cookie to be passed to the server by the client over non-secure channels (HTTP) and
subsequently allows an attacker to e.g. conduct session hijacking attacks.
Detection Method
Checks all cookies sent by the remote HTTP web server /
application over a SSL/TLS connection for a missing 'Secure' cookie attribute.
Details:
[Missing 'Secure' Cookie Attribute (HTTP) OID: 1.3.6.1.4.1.25623.1.0.902661](http://192.168.2.17:9392/nvt/1.3.6.1.4.1.25623.1.0.902661)
Version used:
2023-01-17T10:10:58Z
Affected Software/OS
Any web application accessible via a SSL/TLS connection (HTTPS)
and at the same time also accessible over a cleartext connection (HTTP).
Solution
Solution Type:
Mitigation
Set the 'Secure' cookie attribute for any cookies that are sent
over a SSL/TLS connection.
References
Other
https://www.rfc-editor.org/rfc/rfc6265#section-5.2.5
https://owasp.org/www-community/controls/SecureCookieAttribute
https://wiki.owasp.org/index.php/Testing_for_cookies_attributes_(OTG-SESS-002)

**Comment by juched78** [comment]
Secondary finding:

<html>
<body>
<div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;"><br class="Apple-interchange-newline">Summary</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">The remote HTTP web server / application is missing to set the
'HttpOnly' cookie attribute for one or more sent HTTP cookie.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Detection Result</h2><div style="box-sizing: inherit;"><div class="sc-gRnDUn dLEdfc" style="box-sizing: inherit; min-width: 500px; max-width: 1080px;"><pre class="sc-drKuOJ bOoNFk" style="box-sizing: inherit; white-space: pre-wrap; overflow-wrap: break-word;">The cookies:

Set-Cookie: PHPSESSID=***replaced***; path=/

are missing the "HttpOnly" attribute.</pre></div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Insight</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">The flaw exists if a session cookie is not using the 'HttpOnly'
cookie attribute.

This allows a cookie to be accessed by JavaScript which could lead to session hijacking
attacks.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Detection Method</h2><div style="box-sizing: inherit;"><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">Checks all cookies sent by the remote HTTP web server /
application for a missing 'HttpOnly' cookie attribute.</div>

Details: | Missing 'HttpOnly' Cookie Attribute (HTTP) OID: 1.3.6.1.4.1.25623.1.0.105925
-- | --
Version used: | 2023-01-11T10:12:37Z

</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Affected Software/OS</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">Any web application with session handling in cookies.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Solution</h2><div style="box-sizing: inherit;"><div class="sc-bdVaJa sc-htpNat iKyFgt" style="box-sizing: inherit; flex-direction: row; -webkit-box-pack: start; justify-content: start; -webkit-box-align: center; align-items: center; display: inline-flex;"><div margin="5px" class="sc-bdVaJa sc-bwzfXH kCHBV" style="box-sizing: inherit; display: flex; flex-direction: row; -webkit-box-pack: start; justify-content: start; -webkit-box-align: center; align-items: center; margin-left: -5px;"><b style="box-sizing: inherit; display: inline-flex; margin-left: 5px;">Solution Type:</b><div class="sc-bdVaJa sc-htpNat iKyFgt" style="box-sizing: inherit; display: inline-flex; margin-left: 5px; flex-direction: row; -webkit-box-pack: start; justify-content: start; -webkit-box-align: center; align-items: center;"><div margin="5px" class="sc-bdVaJa sc-bwzfXH kAIqmE" style="box-sizing: inherit; display: flex; flex-direction: row; -webkit-box-pack: start; justify-content: flex-start; -webkit-box-align: center; align-items: center; margin-left: -5px;"><span alt="Mitigation" class="sc-ifAKCX gPHCyz sc-EHOje geAzrQ" data-testid="svg-icon" title="Mitigation" style="box-sizing: inherit; display: inline-flex; margin-left: 5px; height: 16px; width: 16px; line-height: 16px;"><svg width="256" height="256" viewBox="0 0 67.733 67.733" xmlns="http://www.w3.org/2000/svg"><g transform="translate(0 -229.27)"><g transform="matrix(2.0878 0 0 2.2337 41.673 -337.41)"><path class="gui_icon_class" transform="matrix(.12673 0 0 .11845 -19.96 253.7)" d="m96 32-48 48 48 48v-39.928c53.333 0.7429 106.67 9.0706 160 31.928v-16c-53.333-22.857-106.67-31.185-160-31.928v-40.072zm64 96v39.928c-53.333-0.7429-106.67-9.0706-160-31.928v16c53.333 22.857 106.67 31.185 160 31.928v40.072l48-48-48-48z"></path></g></g></svg></span><span style="box-sizing: inherit; display: inline-flex; margin-left: 5px;">Mitigation</span></div></div></div></div><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">Set the 'HttpOnly' attribute for any session cookie.</div></div></div>
</body>
</html>Summary
The remote HTTP web server / application is missing to set the
'HttpOnly' cookie attribute for one or more sent HTTP cookie.
Detection Result
The cookies:

Set-Cookie: PHPSESSID=***replaced***; path=/

are missing the "HttpOnly" attribute.
Insight
The flaw exists if a session cookie is not using the 'HttpOnly'
cookie attribute.

This allows a cookie to be accessed by JavaScript which could lead to session hijacking
attacks.
Detection Method
Checks all cookies sent by the remote HTTP web server /
application for a missing 'HttpOnly' cookie attribute.
Details:
[Missing 'HttpOnly' Cookie Attribute (HTTP) OID: 1.3.6.1.4.1.25623.1.0.105925](http://192.168.2.17:9392/nvt/1.3.6.1.4.1.25623.1.0.105925)
Version used:
2023-01-11T10:12:37Z
Affected Software/OS
Any web application with session handling in cookies.
Solution
Solution Type:
Mitigation
Set the 'HttpOnly' attribute for any session cookie.

**Comment by juched78** [comment]
Another finding: 

<html>
<body>
<div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Summary</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">The remote web server supports the TRACE and/or TRACK
methods. TRACE and TRACK are HTTP methods which are used to debug web server connections.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Detection Result</h2><div style="box-sizing: inherit;"><div class="sc-gRnDUn dLEdfc" style="box-sizing: inherit; min-width: 500px; max-width: 1080px;"><pre class="sc-drKuOJ bOoNFk" style="box-sizing: inherit; white-space: pre-wrap; overflow-wrap: break-word;">The web server has the following HTTP methods enabled: TRACE</pre></div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Insight</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">It has been shown that web servers supporting this methods
are subject to cross-site-scripting attacks, dubbed XST for Cross-Site-Tracing, when used in
conjunction with various weaknesses in browsers.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Detection Method</h2><div style="box-sizing: inherit;"><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">Checks if HTTP methods such as TRACE and TRACK are
enabled and can be used.</div>

Details: | HTTP Debugging Methods (TRACE/TRACK) Enabled OID: 1.3.6.1.4.1.25623.1.0.11213
-- | --
Version used: | 2022-05-12T09:32:01Z

</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Affected Software/OS</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">Web servers with enabled TRACE and/or TRACK methods.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Impact</h2><div style="box-sizing: inherit;"><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">An attacker may use this flaw to trick your legitimate web
users to give him their credentials.</div></div></div><div class="sc-bdVaJa sUcSo" style="box-sizing: inherit; display: flex; flex-direction: column; -webkit-box-pack: center; justify-content: center; -webkit-box-align: stretch; align-items: stretch; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h2 style="box-sizing: inherit; font-size: 18px;">Solution</h2><div style="box-sizing: inherit;"><div class="sc-bdVaJa sc-htpNat iKyFgt" style="box-sizing: inherit; flex-direction: row; -webkit-box-pack: start; justify-content: start; -webkit-box-align: center; align-items: center; display: inline-flex;"><div margin="5px" class="sc-bdVaJa sc-bwzfXH kCHBV" style="box-sizing: inherit; display: flex; flex-direction: row; -webkit-box-pack: start; justify-content: start; -webkit-box-align: center; align-items: center; margin-left: -5px;"><b style="box-sizing: inherit; display: inline-flex; margin-left: 5px;">Solution Type:</b><div class="sc-bdVaJa sc-htpNat iKyFgt" style="box-sizing: inherit; display: inline-flex; margin-left: 5px; flex-direction: row; -webkit-box-pack: start; justify-content: start; -webkit-box-align: center; align-items: center;"><div margin="5px" class="sc-bdVaJa sc-bwzfXH kAIqmE" style="box-sizing: inherit; display: flex; flex-direction: row; -webkit-box-pack: start; justify-content: flex-start; -webkit-box-align: center; align-items: center; margin-left: -5px;"><span alt="Mitigation" class="sc-ifAKCX gPHCyz sc-EHOje geAzrQ" data-testid="svg-icon" title="Mitigation" style="box-sizing: inherit; display: inline-flex; margin-left: 5px; height: 16px; width: 16px; line-height: 16px;"><svg width="256" height="256" viewBox="0 0 67.733 67.733" xmlns="http://www.w3.org/2000/svg"><g transform="translate(0 -229.27)"><g transform="matrix(2.0878 0 0 2.2337 41.673 -337.41)"><path class="gui_icon_class" transform="matrix(.12673 0 0 .11845 -19.96 253.7)" d="m96 32-48 48 48 48v-39.928c53.333 0.7429 106.67 9.0706 160 31.928v-16c-53.333-22.857-106.67-31.185-160-31.928v-40.072zm64 96v39.928c-53.333-0.7429-106.67-9.0706-160-31.928v16c53.333 22.857 106.67 31.185 160 31.928v40.072l48-48-48-48z"></path></g></g></svg></span><span style="box-sizing: inherit; display: inline-flex; margin-left: 5px;">Mitigation</span></div></div></div></div><div class="sc-iomxrj juboHc" style="box-sizing: inherit; white-space: pre-line;">Disable the TRACE and TRACK methods in your web server
configuration.

Please see the manual of your web server or the references for more information.</div></div></div>
</body>
</html>Summary
The remote web server supports the TRACE and/or TRACK
methods. TRACE and TRACK are HTTP methods which are used to debug web server connections.
Detection Result
The web server has the following HTTP methods enabled: TRACE
Insight
It has been shown that web servers supporting this methods
are subject to cross-site-scripting attacks, dubbed XST for Cross-Site-Tracing, when used in
conjunction with various weaknesses in browsers.
Detection Method
Checks if HTTP methods such as TRACE and TRACK are
enabled and can be used.
Details:
[HTTP Debugging Methods (TRACE/TRACK) Enabled OID: 1.3.6.1.4.1.25623.1.0.11213](http://192.168.2.17:9392/nvt/1.3.6.1.4.1.25623.1.0.11213)
Version used:
2022-05-12T09:32:01Z
Affected Software/OS
Web servers with enabled TRACE and/or TRACK methods.
Impact
An attacker may use this flaw to trick your legitimate web
users to give him their credentials.
Solution
Solution Type:
Mitigation
Disable the TRACE and TRACK methods in your web server
configuration.

Please see the manual of your web server or the references for more information.

**Comment by szaimen** [comment, marked as answer]
Hi, the problem arises because you scanned port 11001 without TLS. You need to scan  the Reverse Proxy connection. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
